grc/core/generator/top_block.py

```diff
@@ -234,15 +234,20 @@ def make_port_sig(port):
 
         # Get the virtual blocks and resolve their connections
         connection_factory = fg.parent_platform.Connection
-        virtual = [c for c in connections if isinstance(c.source_block, blocks.VirtualSource)]
-        for connection in virtual:
+        virtual_source_connections = [c for c in connections if isinstance(c.source_block, blocks.VirtualSource)]
+        virtual_sink_connections = [c for c in connections if isinstance(c.sink_block, blocks.VirtualSink)]
+
+        for connection in virtual_source_connections:
             sink = connection.sink_port
             for source in connection.source_port.resolve_virtual_source():
                 resolved = connection_factory(fg.orignal_flowgraph, source, sink)
                 connections.append(resolved)
             # Remove the virtual connection
             connections.remove(connection)
 
+        for connection in virtual_sink_connections:
+            connections.remove(connection)
```

Comment:
This prevents lines like `self.connect((self.analog_sig_source_x_0, 0), (self.virtual_sink_0, 0))` from appearing in the Python files. Attempting to use connect() with a virtual source/sink creates an `AttributeError: 'top_block_sptr' object has no attribute 'virtual_sink_0'`.